LibTCPServer.emx

```diff
@@ -341,38 +345,49 @@ srv->start();
             </trigger>
           </transition>
           <transition xmi:id="_dEMZQO5XEeitL-zKjjNp2A" name="stop" source="_5rHTee5TEeitL-zKjjNp2A" target="_bd0RUO5XEeitL-zKjjNp2A">
+            <effect xmi:type="uml:OpaqueBehavior" xmi:id="_NftnkOeBEe6S2oclZuMFIg" name="Effect">
+              <language><![CDATA[C++]]></language>
+              <body><![CDATA[delete serverThread;
+delete job;]]></body>
+            </effect>
             <trigger xmi:id="_5ngowO5XEeitL-zKjjNp2A" name="" event="_ZJ5G7e5XEeitL-zKjjNp2A" port="_5rHTPO5TEeitL-zKjjNp2A"/>
           </transition>
         </region>
       </ownedBehavior>
       <ownedBehavior xmi:type="uml:OpaqueBehavior" xmi:id="_5rH6Qu5TEeitL-zKjjNp2A" name="handleReceivedMessageDefault" specification="_5rH6Ru5TEeitL-zKjjNp2A">
         <language><![CDATA[C++]]></language>
-        <body><![CDATA[	// First check for some special HTTP requests
+        <body><![CDATA[
+	if(str.empty())
+	{
+		return;
+	}
+	// First check for some special HTTP requests
 	if (str.rfind("GET /getPorts", 0) == 0) {
 		std::string strResult = getPorts();
 		conn->replyHTTPStatus(ITCPConnection::Status::OK, strResult);
 		return;
 	}
 
 	// Everything else is handled as TCP requests
-	Parser parser;
-	Var result;
-	try {
-		result = parser.parse(str);
+	RTJsonParser parse;
+	RTJsonResult jsonResult;
+	if(!parse.parseJsonString(jsonResult, str))
+	{
+		conn->replyHTTPStatus(ITCPConnection::Status::Error, "Failed to parse JSON: " + str);
```

Comment:
I think we should only reply with HTTP status if we get an HTTP request (as above). For a regular TCP request a HTTP reply will just look unreadable due to lots of extra information that we don't need.